usersync/bidderchooser.go

```diff
@@ -0,0 +1,56 @@
+package usersync
+
+// bidderChooser determines which bidders to consider for user syncing.
+type bidderChooser interface {
+	// choose returns an ordered collection of potentially non-unique bidders.
+	choose(requested, available []string, cooperative Cooperative) []string
+}
+
+// standardBidderChooser implements the bidder choosing algorithm per official Prebid specification.
+type standardBidderChooser struct {
+	shuffler shuffler
+}
+
+func (c standardBidderChooser) choose(requested, available []string, cooperative Cooperative) []string {
+	if len(requested) == 0 {
+		return c.shuffledCopy(available)
+	}
+
+	if cooperative.Enabled {
+		return c.chooseCooperative(requested, available, cooperative.PriorityGroups)
+	}
+
+	return c.shuffledCopy(requested)
+}
+
+func (c standardBidderChooser) chooseCooperative(requested, available []string, priorityGroups [][]string) []string {
+	biddersCapacity := int(float64(len(available)) * 1.5)
```

Comment:
Are you setting a length of 1.5 because it is possible the same bidder appears more than once in the list of chosen bidders?

Reply:
Yes, exactly. I'm not sure what to use for a default array size, so I figure there would be some overlap at least with the `available` items. I randomly chose to use a 50% overhead. This is purely a guess to try to avoid array reallocation within the slice.

Reply:
I'll add a comment in the code to explain the 1.5.